app/controllers/recipes_controller.rb

```diff
@@ -21,6 +21,36 @@ def create
     end
   end
 
+  def edit
+    if current_user.id == Recipe.find(params[:id]).user.id
+      @recipe = current_user.recipes.find(params[:id])
+    else
+      flash[:error] = "You are not authorized to edit this recipe."
+      redirect_to root_path
+    end
+  end
+
+  def update
+    if current_user.id == Recipe.find(params[:id]).user.id
+      @recipe = current_user.recipes.find(params[:id])
+      if @recipe.update_attributes(recipe_params)
```

Comment:
Indentation in this method needs work, its a little hard to follow at the moment